apps/system/js/advanced_telemetry.js

```diff
@@ -570,23 +570,21 @@
   // This method loops through the payload of histograms and converts each
   // histogram payload to the 'dict' format required by the server for parsing.
   AdvancedTelemetryPing.prototype.packHistograms = function() {
-    var nameAddonHist = this.packet.payload.addonHistograms;
-    var nameKeyHist = this.packet.payload.keyedHistograms;
-
-    // Pack the Addon Histograms.
-    for (var addon in nameAddonHist) {
-      for (var hist in nameAddonHist[addon]) {
-        nameAddonHist[addon][hist] =
-          this.packHistogram(nameAddonHist[addon][hist]);
-      }
-    }
+    function packHistogramType(level, self) {
+      for (var key in level) {
+        // Discard any Histograms with no data.
+        if (Object.keys(level[key]).length === 0) {
+          delete level[key];
+          continue;
+        }
 
-    // Pack the Keyed Histograms.
-    for (var key in nameKeyHist) {
-      for (var hist2 in nameKeyHist[key]) {
-        nameKeyHist[key][hist2] = this.packHistogram(nameKeyHist[key][hist2]);
+        for (var hist2 in level[key]) {
+          level[key][hist2] = self.packHistogram(level[key][hist2]);
+        }
       }
     }
+    packHistogramType(this.packet.payload.addonHistograms, this);
+    packHistogramType(this.packet.payload.keyedHistograms, this);
```

Comment:
I believe using 'bind' is the preferred way to set `this` (although you're approach is actually more straightforward).